image/image.go

```diff
@@ -787,3 +788,65 @@ func copyLocalSnapFile(snapPath, targetDir string, info *snap.Info) (dstPath str
 	dst := filepath.Join(targetDir, filepath.Base(info.MountFile()))
 	return dst, osutil.CopyFile(snapPath, dst, 0)
 }
+
+func ValidateSeed(seedFile string) error {
+	seed, err := snap.ReadSeedYaml(seedFile)
+	if err != nil {
+		return err
+	}
+
+	// read the snaps info
+	snapInfos := make(map[string]*snap.Info)
+	for _, seedSnap := range seed.Snaps {
+		fn := filepath.Join(filepath.Dir(seedFile), "snaps", seedSnap.File)
+		snapf, err := snap.Open(fn)
+		if err != nil {
+			return err
```

Comment:
if a snap file from the seed is missing, is this really a catastrophic failure? I think we could just append `err` to `errs` (creating it earlier), continue in the loop and not have this snap in the map. 

Something like:

```suggestion
			errs = append(errs,err)
			continue
```

Reply:
I updated the PR, I could not apply the suggestions directly as its slightly more complicated but I also added tests so things should be fine. The suggestion is great!